custom/icds_reports/views.py

```diff
@@ -884,6 +884,15 @@ def post(self, request, *args, **kwargs):
 
         location = request.POST.get('location', '')
 
+        if not location and not request.couch_user.has_permission(
+                self.kwargs['domain'], 'access_all_locations'
+        ):
+            return HttpResponseBadRequest()
```

Comment:
nitpick: wouldn't a 403 make more sense for both of these? that's what `location_restricted_response` does.

Reply:
Yeah, that sounds right. Fixed